tests/src/python/test_provider_shapefile.py

```diff
@@ -1142,6 +1142,18 @@ def testFilterWithComment(self):
         f = next(vl.getFeatures())
         self.assertEqual(f['name'], 'Apple')
 
+    def testRecomputeExtent(self):
+        """Test shrink extent correctly"""
+
+        vl = self.getEditableLayer()
+        extent_area = vl.extent().area()
+        vl.startEditing()
+        for fet in vl.getFeatures():
+            vl.translateFeature(fet.id(), -1, -1)
+        vl.commitChanges()
+        self.assertEqual(vl.extent().area(), extent_area)
```

Comment:
This test isn't super convincing. You should really test that the extent has been updated (if it hasn't been updated, the area will also be preserved). And it should also test that the extent is OK after closing and re-opening the dataset

Reply:
Thank you for pointing this out.